src/hotspot/share/opto/type.hpp

```diff
@@ -216,15 +216,15 @@ class Type {
   // Create a new hash-consd type
   static const Type *make(enum TYPES);
   // Test for equivalence of types
-  static int cmp( const Type *const t1, const Type *const t2 );
+  static bool cmp(const Type* t1, const Type* t2);
   // Test for higher or equal in lattice
   // Variant that drops the speculative part of the types
-  bool higher_equal(const Type *t) const {
-    return !cmp(meet(t),t->remove_speculative());
+  bool higher_equal(const Type* t) const {
+    return cmp(meet(t), t->remove_speculative());
```

Comment:
Also: can you explain the comment above the method here? If it is true that `cmp` only gave back `0 or 1`, and now only a bool, then how does that tell us anything about "higher" or "lower" in the lattice? Are we not constrained to equal or not equal, where if we have not equal, we don't know if it is higher or lower?

Reply:
Never mind, I overlooked the `meet` in there... ignore this,